anchor-reference-server/src/main/java/org/stellar/anchor/reference/service/FeeService.java

```diff
@@ -0,0 +1,26 @@
+package org.stellar.anchor.reference.service;
+
+import java.math.BigDecimal;
+import org.springframework.stereotype.Service;
+import org.stellar.anchor.exception.BadRequestException;
+import org.stellar.platform.apis.callbacks.requests.GetFeeRequest;
+import org.stellar.platform.apis.callbacks.responses.GetFeeResponse;
+import org.stellar.platform.apis.shared.Amount;
+
+@Service
+public class FeeService {
+  BigDecimal feePercent = new BigDecimal("0.02"); // fixed 2% fee.
+  BigDecimal feeFixed = new BigDecimal("0.1");
+
+  public GetFeeResponse getFee(GetFeeRequest request) throws BadRequestException {
```

Comment:
Maybe we don't need to implement this now, but ultimately we want to check here that the sending and receiving customers are accepted for their associated types and return an error response if they aren't.

Reply:
This is part of reference. I think we can discuss the scenario that we want to do in the reference. 

I thought again about overloading `/fee` for the purpose of checking the sending and receiving customer. The benefit of saving the extra call is at the cost of some confusion to the anchor. We should probably discuss this during our on-site next week.

Reply:
I had put some TODO: tags as reminder.

Reply:
The description on the `GET /fee` endpoint is:

> The Platform will make this request to the anchor every time a transaction is initiated by a client application.
>  
> **The Anchor must ensure that the sending and receiving customers specified in the request meet the KYC requirements necessary to partake in the transaction described in the request.**
> 
> The anchor must return the fee it will charge for the transaction.

Unless we add another endpoint for verifying the transaction itself, this is where anchors should verify senders and receivers.